wrangler-api/src/main/java/io/cdap/wrangler/api/Directive.java

```diff
@@ -126,4 +129,11 @@ default List<EntityCountMetric> getCountMetrics() {
     // no op
     return null;
   }
+
+  @Override
+  default Relation transform(RelationalTranformContext relationalTranformContext,
+      Relation relation) {
+    // no-op
```

Comment:
Looking just at this class, it seems like every existing directive would become a no-op in SQL, which would be incorrect. It only makes sense when seeing the isSQLSupported() method in the RelationalTransform interface. 

I think it would be cleaner if the Directive interface did not extend the RelationalTransform interface, with only directives that do support SQL implementing it. Then the isSQLSupported() method is not needed, Wrangler just checks if the directive is an instance of RelationTransform. This way it is impossible to implement a directive where the transform() and isSQLSupported() methods don't match up.